controls/aws-foundations-cis-2.1.4.rb

```diff
@@ -0,0 +1,216 @@
+
+**From Command Line:**
+
+To check Public access
+settings for this account status, run the following command,
+`aws s3control
+get-public-access-block --account-id <ACCT_ID> --region <REGION_NAME>`
+
+Output if
+Block Public access is enabled:
+
+```
+{
+ \"PublicAccessBlockConfiguration\": {
+
+\"IgnorePublicAcls\": true,
+ \"BlockPublicPolicy\": true,
+ \"BlockPublicAcls\": true,
+
+\"RestrictPublicBuckets\": true
+ }
+}
+```
+
+If the output reads `false` for the
+separate configuration settings then proceed to the remediation. "
+  desc 'fix',
+       "**If utilizing Block Public Access (bucket settings)**
+
+**From Console:**
+
+1. Login
+to AWS Management Console and open the Amazon S3 console using
+https://console.aws.amazon.com/s3/
+2. Select the Check box next to the Bucket.
+3.
+Click on 'Edit public access settings'.
+4. Click 'Block all public access'
+5. Repeat for
+all the buckets in your AWS account that contain sensitive data.
+
+**From Command
+Line:**
+
+1. List all of the S3 Buckets
+```
+aws s3 ls
+```
+2. Set the Block Public
+Access to true on that bucket
+```
+aws s3api put-public-access-block --bucket
+<name-of-bucket> --public-access-block-configuration \"BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true\"
+```
+
+**If
+utilizing Block Public Access (account settings)**
+
+**From Console:**
+
+If the
+output reads `true` for the separate configuration settings then it is set on the
+account.
+
+1. Login to AWS Management Console and open the Amazon S3 console using
+https://console.aws.amazon.com/s3/
+2. Choose `Block Public Access (account
+settings)`
+3. Choose `Edit` to change the block public access settings for all the buckets
+in your AWS account
+4. Choose the settings you want to change, and then choose `Save`. For
+details about each setting, pause on the `i` icons.
+5. When you're asked for confirmation,
+enter `confirm`. Then Click `Confirm` to save your changes.
+
+**From Command
+Line:**
+
+To set Block Public access settings for this account, run the following
+command:
+```
+aws s3control
+put-public-access-block
+--public-access-block-configuration
+BlockPublicAcls=true, IgnorePublicAcls=true, BlockPublicPolicy=true,
+RestrictPublicBuckets=true
+--account-id <value>
+``` "
+  desc 'impact',
+       "When you apply Block Public Access settings to an account, the settings apply to all AWS
+Regions globally. The settings might not take effect in all Regions immediately or
+simultaneously, but they eventually propagate to all Regions. "
+  impact 0.5
+  ref 'https://docs.aws.amazon.com/AmazonS3/latest/user-guide/block-public-access-account.html'
+  tag nist: ['AC-3']
+  tag severity: 'medium '
+  tag cis_controls: [{ '8' => ['3.3'] }]
+
+  exempt_buckets = input('exempt_buckets')
+  s3_buckets = aws_s3_buckets.bucket_names
+  failing_buckets = []
+
+  describe 'No Tests Defined Yet' do
+    skip 'No Tests have been written for this control yet'
+  end
```

Comment:
We should perhaps make this a bit more clean so they know which buckets have 'by bucket' restrictions and 'by account' restrictions.